03-understanding-survey-data-files.Rmd

```diff
@@ -1 +1,238 @@
+
+The document provides more information about the variables, summarized below:
+
+For weight | Use variance unit/PSU/cluster | and use variance stratum
+-----------|-------------------------------|-------------------------
+V200010a| V200010c| V200010d
+V200010b| V200010c| V200010d
+
+As mentioned above, we want to conduct ANES data analysis with weights to accurately represent the population. The user guide references a supplemental document called "How to Analyze ANES Survey Data"^[DeBell, Matthew. 2010. How to Analyze ANES Survey Data. ANES Technical Report Series no. nes012492. Palo Alto, CA, and Ann Arbor, MI: Stanford University and the University of Michigan] as a 'how-to guide' to help us with our analysis.
+
+Recall the "Sample Design and Respondent Recruitment" section:
+
+> The target population for the fresh cross-section was the 231 million non-institutional U.S. citizens aged 18 or older living in the 50 US states or the District of Columbia.
+
+We will use Current Population Survey (CPS) to find a number of the non-institutional U.S. citizens aged 18 or older living in the 50 U.S. states or D.C. The {censusapi} package allows us to do this in a reproducible way:
+
+``` r
+library(censusapi)
+library(tidyverse)
+
+cps_state_in <- getCensus(
+  name = "cps/basic/nov",
+  vintage = 2020,
+  region = "state",
+  vars = c("HRHHID", "HRMONTH", "HRYEAR4", "PRTAGE", "PRCITSHP", "PWSSWGT"),
+  key = Sys.getenv("CENSUS_API_KEY")
+)
+
+cps_state <- cps_state_in %>%
+  as_tibble() %>%
+  mutate(across(.fns = as.numeric))
+
+# Confirm this doesn't include territories
+cps_state %>%
+  count(state)
+```
+
+    ## # A tibble: 51 x 2
+    ##    state     n
+    ##    <dbl> <int>
+    ##  1     1  2406
+    ##  2     2  1289
+    ##  3     4  1969
+    ##  4     5  1988
+    ##  5     6  9574
+    ##  6     8  1365
+    ##  7     9  1157
+    ##  8    10  1285
+    ##  9    11  1622
+    ## 10    12  5055
+    ## # ... with 41 more rows
+
+``` r
+# Confirm this is only November 2020
+cps_state %>%
+  count(HRMONTH, HRYEAR4)
+```
+
+    ## # A tibble: 1 x 3
+    ##   HRMONTH HRYEAR4      n
+    ##     <dbl>   <dbl>  <int>
+    ## 1      11    2020 112037
+
+``` r
+# Voting age citizen population
+targetpop <- cps_state %>%
+  as_tibble() %>%
+  filter(PRTAGE >= 18,
+         PRCITSHP %in% (1:4)) %>%
+  pull(PWSSWGT) %>%
+  sum()
+```
+
+The target population in 2020 is 231,592,693. This information gives us what we need to create the post-election survey object with {srvyr}:
+
+```{r}
+#| eval: FALSE
+library(tidyverse)
+library(here)
+library(srvyr)
+
+anes <- read_rds(here("Data", "anes_2020.rds")) %>%
+  mutate(Weight = V200010b / sum(V200010b) * 231592693)
+
+anes_des <- anes %>%
+  as_survey_design(
+    weights = Weight,
+    strata = V200010d,
+    ids = V200010c,
+    nest = TRUE
+  )
+
+summary(anes_des)
+```
```

Comment:
Is there any reason this code isn't actually run? I think we should and show that the value of targetpop=231592693

Reply:
I think it was because of not having the data in the folders and also not having a Census key :D I am adding both now.